scripts/java/hilla-scripts/src/main/java/com/vaadin/hilla/scripts/FlowPackageJsonUpdater.java

```diff
@@ -0,0 +1,154 @@
+            logger().debug("Applying Flow {} package.json.", id);
+        }
+        var name = String.join("/", FRONTEND_RESOURCES_PATH, "dependencies", id,
+                "package.json");
+        var resource = getClass().getClassLoader().getResource(name);
+        try {
+            assert resource != null;
+            var flowPackageJsonTree = JacksonUtils.readTree(
+                    IOUtils.toString(resource, StandardCharsets.UTF_8));
+            Stream.of(DEPENDENCIES, DEV_DEPENDENCIES).forEach((section) -> {
+                // NOTE: Some dependencies are devDependencies in Hilla,
+                // such as "react". Hence we update at "devDependencies".
+                updateTreeAt(DEV_DEPENDENCIES, flowPackageJsonTree.at(section));
+            });
+        } catch (IOException e) {
+            logger().error(
+                    "Unable to read Flow {} package.json resource, skipping.",
+                    id);
+        }
+    }
+
+    private void updateTreeAt(JsonPointer pointer, JsonNode value) {
+        var current = tree.at(pointer);
+        if (current.isMissingNode()) {
+            if (logger().isDebugEnabled()) {
+                logger().debug(
+                        "Skipping update for {}, missing from the Hilla package.json file.",
+                        pointer);
+            }
+            return;
+        }
+
+        if (value instanceof ObjectNode valueObject) {
+            valueObject.fields().forEachRemaining((field) -> {
+                updateTreeAt(pointer.appendProperty(field.getKey()),
+                        field.getValue());
+            });
+            return;
+        }
+
+        if (value.equals(current)) {
+            if (logger().isDebugEnabled()) {
+                logger().debug("Skipping update for {}, same value of {}.",
+                        pointer, value);
+            }
+            return;
+        }
+
+        if (logger().isDebugEnabled()) {
+            logger().debug("Updating {} from {} to {}.", pointer, current,
+                    value);
+        }
+        var parentTree = tree.at(pointer.head());
+        if (parentTree instanceof ObjectNode parentTreeObject) {
+            parentTreeObject.set(pointer.last().getMatchingProperty(), value);
+        }
+    }
+
+    private void saveChanges() throws IOException {
+        var objectMapper = JacksonUtils.getMapper();
+        var filePrinter = new DefaultPrettyPrinter()
+                .withSeparators(Separators.createDefaultInstance()
+                        .withObjectFieldValueSpacing(Separators.Spacing.AFTER))
+                .withArrayIndenter(DefaultIndenter.SYSTEM_LINEFEED_INSTANCE);
+        var contents = objectMapper.writer().with(filePrinter)
+                .writeValueAsString(tree) + DefaultIndenter.SYS_LF;
+        FileIOUtils.writeIfChanged(packageJsonFile.toFile(), contents);
+    }
+
+    public static void main(String[] args)
+            throws URISyntaxException, IOException {
```

Comment:
`URISyntaxException` doesn't seem to be thrown, so
```suggestion 
            throws IOException {
```